From bc384f922b0dc49073a43d4f6fa212df49f688b6 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@gmail.com>
Date: Mon, 26 Sep 2016 07:41:49 +1000
Subject: [PATCH] Enable stateless configuration

Set install path to use (KDE_INSTALL_)DBUSDIR from extra-cmake-modules.
Edit KF5Auth macro so that other packages point to the same path.
---
 cmake/KF5AuthMacros.cmake | 2 +-
 src/ConfigureChecks.cmake | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/KF5AuthMacros.cmake b/cmake/KF5AuthMacros.cmake
index 15f05c7..419618f 100644
--- a/cmake/KF5AuthMacros.cmake
+++ b/cmake/KF5AuthMacros.cmake
@@ -37,7 +37,7 @@ function(KAUTH_INSTALL_HELPER_FILES HELPER_TARGET HELPER_ID HELPER_USER)
         configure_file(${KAUTH_STUB_FILES_DIR}/dbus_policy.stub
                         ${CMAKE_CURRENT_BINARY_DIR}/${HELPER_ID}.conf)
         install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${HELPER_ID}.conf
-                DESTINATION ${SYSCONF_INSTALL_DIR}/dbus-1/system.d/)
+                DESTINATION ${KDE_INSTALL_DBUSDIR}/system.d/)

         configure_file(${KAUTH_STUB_FILES_DIR}/dbus_service.stub
                         ${CMAKE_CURRENT_BINARY_DIR}/${HELPER_ID}.service)
diff --git a/src/ConfigureChecks.cmake b/src/ConfigureChecks.cmake
index d46761a..f08eb30 100644
--- a/src/ConfigureChecks.cmake
+++ b/src/ConfigureChecks.cmake
@@ -222,7 +222,7 @@ if(KAUTH_HELPER_BACKEND_NAME STREQUAL "DBUS")

     # Install some files as well
     install( FILES backends/dbus/org.kde.kf5auth.conf
-             DESTINATION ${SYSCONF_INSTALL_DIR}/dbus-1/system.d )
+             DESTINATION ${KDE_INSTALL_DBUSDIR}/system.d )

     install( FILES backends/dbus/dbus_policy.stub
                    backends/dbus/dbus_service.stub
--
2.10.0
